llvm/include/llvm/MC/MCAssembler.h

```diff
@@ -279,6 +280,10 @@ class MCAssembler {
   unsigned getELFHeaderEFlags() const { return ELFHeaderEFlags; }
   void setELFHeaderEFlags(unsigned Flags) { ELFHeaderEFlags = Flags; }
 
+  /// ELF e_ident[EI_ABIVERSION] value
+  unsigned char getELFHeaderABIVersion() const { return ELFHeaderABIVersion; }
```

Comment:
See `llvm/lib/MC/MCELFStreamer.cpp:Asm.getWriter().markGnuAbi()`. This can be moved to `ELFObjectWriter` and revert the `MCELFObjectTargetWriter` change below.

Reply:
Sure, I can do it like that if you prefer. Would you mind verifying the update is what you had in mind?